### ISSUE TITLE
Making it work in Windows 10

### ISSUE THREAD

**Issue by johnRivs** [calebporzio/better-phpunit#35]
I was able to open a terminal and manually do what Better PHPUnit does, but I can't translate that into the config file.
```
C:\Program Files (x86)\cmder\vendor\git-for-windows\usr\bin\ssh.exe -tt -p2222 vagrant@localhost -i "C:/Homestead/.vagrant/machines/homestead-7/virtualbox/private_key" -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o PasswordAuthentication=no -o IdentitiesOnly=yes -o LogLevel=FATAL -o ForwardAgent=yes "/home/vagrant/www/spotify/vendor/bin/phpunit /home/vagrant/www/spotify/tests/Feature/Spotify/APITest.php --filter it_builds_an_api_client"
```
`C:\Program Files (x86)\cmder\vendor\git-for-windows\usr\bin\ssh.exe`: Path to the SSH executable.
`-tt`: No idea but Better PHPUnit uses it.
`-p2222`: Port, according to `vagrant ssh-config`.
`vagrant`: User.
`localhost`: Host.
`-i "C:/Homestead/.vagrant/machines/homestead-7/virtualbox/private_key" -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o PasswordAuthentication=no -o IdentitiesOnly=yes -o LogLevel=FATAL -o ForwardAgent=yes`: More stuff from `vagrant ssh-config` so no password is asked. Pretty sure you don't need everything though.

Now, when it comes to the `phpunit` path and the test file, it doesn't point to the path within the virtual machine, instead it tries to find it on the host machine and fails, like `c:/path/to/project/vendor/bin/phpunit` which doesn't exist. Also, I have to add all those flags as suffix which doesn't sit well with PHPUnit because it thinks I'm trying to pass all that to it: `unrecognized option -- o`.

1. How can I make Better PHPUnit find the test files within Homestead?
2. Is there a way to swap flags and the `phpunit` command?

**Comment by hagabaka** [comment]
You can map paths in the setting `better-phpunit.ssh.paths`:
```
{
  "better-phpunit.ssh.paths": {
    "c:/Users/myname/Projects/foo": "/home/vagrant/code/foo"
  }
}
```
The Windows path is treated case-sensitively even though it shouldn't be. For example using `C:` instead of `c:` will cause the path mapping to be ignored.

You can put the ssh parameters in the `better-phpunit.ssh.binary` setting after the command. At least that works for me. For paths containing spaces, the only solution I've found is to use the "8.3" name, for example `"C:/Progra~1/"` instead of `"C:/Program Files/"`.

The `-tt` option is hard-coded, and I think it's supposed to force ssh to use TTY, but it causes PuTTY to exit.

**Comment by johnRivs** [comment]
I don't get it. Why do I need any Windows path at all when it comes to executing `phpunit`? Once the command has SSH'd into the virtual machine, it's all Linux.

Using `better-phpunit.ssh.binary` worked to swap the flags and the `phpunit` command.

Spaces shouldn't be a problem, since this package wraps the entire command with quotes. It does work on the example I gave at the beginning.

**Comment by hagabaka** [comment]
I assume you are running the extension on Windows, so it only sees the files in Windows, including the test case and `phpunit`. However when running tests through ssh, it translates from your local paths to remote paths using the `better-phpunit.ssh.paths` setting.

**Comment by johnRivs** [comment]
Yeah, I'm looking at the source code and that's what I thought. Now I'm trying to figure out if the space is messing things up because all my project paths start with `C:\Google Drive`.

**Comment by johnRivs** [comment]
Ok, I got the paths right:
```
"better-phpunit.ssh.paths": {
        "c:/Google\\ Drive/Web/www": "/home/vagrant/www"
    }
```

But now I get this:
```
> Executing task: C:\PROGRA~2\cmder\vendor\git-for-windows\usr\bin\ssh.exe -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o PasswordAuthentication=no -o IdentitiesOnly=yes -o LogLevel=FATAL -o ForwardAgent=yes -i C:/Homestead/.vagrant/machines/homestead-7/virtualbox/private_key -tt -p2222 vagrant@localhost "/home/vagrant/www/spotify/vendor/bin/phpunit /home/vagrant/www/spotify/tests/Feature/Spotify/APITest.php --filter '^.*::it_builds_an_api_client$' --stop-on-failure" <

The system cannot find the path specified.
```

**Comment by calebporzio** [comment]
This should be fixed by #45 -> upgrade to version 0.8
